src/sweeper/sweepers/duplicates.py

```diff
@@ -106,18 +106,32 @@ def try_fix(self):
 
         log.info(f'Workspace is:   {self.workspace}')
         with arcpy.EnvManager(workspace=self.workspace):
-            try:
-                duplicate_features = arcpy.management.MakeFeatureLayer(self.table_name, temp_feature_layer, sql)
-
-                log.info(f'attempting to delete {len(self.oids_with_issues)} duplicate records')
-
-                arcpy.management.DeleteFeatures(duplicate_features)
-            except Exception as error:
-                error_message = f'unable to delete features {error}'
-                report['issues'].append(error_message)
-            finally:
-                if arcpy.Exists(temp_feature_layer):
-                    arcpy.management.Delete(temp_feature_layer)
+            if self.is_table:
+                try:
+                    duplicate_features = arcpy.management.MakeTableView(self.table_name, temp_feature_layer, sql)
+
+                    log.info(f'attempting to delete {len(self.oids_with_issues)} duplicate records')
+
+                    arcpy.management.DeleteRows(duplicate_features)
+                except Exception as error:
+                    error_message = f'unable to delete features {error}'
+                    report['issues'].append(error_message)
+                finally:
+                    if arcpy.Exists(temp_feature_layer):
+                        arcpy.management.Delete(temp_feature_layer)
```

Comment:
All of this code could be removed if we act on the output from make table or feature right?

Reply:
Hows this: beffcb129866daac969c93b19d52501f8bff57dd.  I don't think we can get rid of everything because we have to call different arcpy tools for tables and feature classes.